.github/workflows/dev_nifi.yaml

```diff
@@ -0,0 +1,110 @@
+---
+name: Build and publish NiFi for signal-processing demo
+
+env:
+  IMAGE_NAME: nifi
+  IMAGE_VERSION: 1.27.0-postgresql
+  REGISTRY_PATH: stackable
+  DOCKERFILE_PATH: "../demos/signal-processing/Dockerfile-nifi"
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - demos/signal-processing/Dockerfile-nifi
+      - .github/workflows/dev_nifi.yaml
+
+jobs:
+  build:
+    name: Build/Publish ${{ matrix.runner.arch }} Image
+    permissions:
+      id-token: write
+    runs-on: ${{ matrix.runner.name }}
+    strategy:
+      matrix:
+        runner:
+          - {name: "ubuntu-latest", arch: "amd64"}
+          - {name: "ubicloud-standard-8-arm", arch: "arm64"}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - name: Free Disk Space (Ubuntu)
+        uses: jlumbroso/free-disk-space@54081f138730dfa15788a46383842cd2f914a1be # v1.3.1
+        with:
+          # This might remove tools that are actually needed, if set to "true" but
+          # frees about 6 GB.
+          tool-cache: false
+
+          # All of these default to true, but feel free to set to "false" if
+          # necessary for your workflow.
+          android: true
+          dotnet: true
+          haskell: true
+          large-packages: true
+          docker-images: true
+          swap-storage: true
+
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@988b5a0280414f521da01fcc63a27aeeb4b104db # v3.6.1
+
+      - name: Build image
+        env:
+          IMAGE_REPOSITORY: ${{ env.IMAGE_NAME }}
+          IMAGE_ARCH: ${{ matrix.runner.arch }}
+        shell: bash
+        run: |
+          set -euo pipefail
+          echo "Building ${{ env.IMAGE_NAME }}"
+          docker buildx build -f "${{ env.DOCKERFILE_PATH }}" --platform "linux/${{ env.IMAGE_ARCH }}" --tag "localhost/${{ env.IMAGE_NAME }}:${{ env.IMAGE_VERSION }}" .
+          echo "IMAGE_ARCH=${{ env.IMAGE_ARCH }}" >> "$GITHUB_ENV"
```

Comment:
This can soon be replaced by a new action for general image builds: https://github.com/stackabletech/actions/pull/3

Reply:
It is now available: https://github.com/stackabletech/actions/releases/tag/0.0.3

Reply:
That was fast :smile: thank you! Then I'll try it out immediately!

Reply:
I'm also using it, so here is a snippet that might help:

```yaml
      - name: Build Product Image
        id: build
        uses: stackabletech/actions/build-container-image@c179c509951343185095cbf032248b71ee6998f4 # 0.0.3
        with:
          image-name: nifi
          image-index-manifest-tag: ${{ env.IMAGE_VERSION }}
          container-file: ${{ env.DOCKERFILE_PATH }}
```

You can then use the tag from the output for the publish steps

Reply:
https://github.com/stackabletech/demos/pull/103/commits/9217e297e5a3335f93a4e775bfa9991f2e9799fa :) feel free to take a look if it looks fine